src/latex/generator.ml

```diff
@@ -283,9 +284,24 @@ let rec block ~in_source (l : Block.t) =
   let one (t : Block.one) =
     match t.desc with
     | Inline i -> inline ~verbatim:false ~in_source:false i
-    | Audio (_, content) | Video (_, content) | Image (_, content) ->
-        txt ~verbatim:false ~in_source:false [ content ]
+    | Image (Internal (Resolved x), alt) ->
+      let dir, file = Link.get_dir_and_file x.page in
+      begin match Fpath.(get_ext @@ v file) with
+      | "" | ".jpeg" | ".png" | ".pdf" ->
+        let fpath = Fpath.v (String.concat Fpath.dir_sep (dir @ [file])) in
+        [ Image fpath ]
+      | _ ->
+        txt ~verbatim:false ~in_source:false [ alt ]
         @ if in_source then [] else [ Break Paragraph ]
+      end
+    | Image (Internal Unresolved, c) | Audio(Internal _, c) | Video (Internal _, c) ->
+        txt ~verbatim:false ~in_source:false [ c ]
+        @ if in_source then [] else [ Break Paragraph ]
```

Comment:
Nitpick: using a `when` clause removes the duplication of the "alternative text" usage:

```suggestion
    | Image (Internal (Resolved x), alt)
      when List.mem Fpath.(get_ext @@ v file) [ ""; ".jpeg"; ".png"; ".pdf" ] ->
        let dir, file = Link.get_dir_and_file x.page in
        let fpath = Fpath.v (String.concat Fpath.dir_sep (dir @ [ file ])) in
        [ Image fpath ]
    | Image (Internal _, c) | Audio (Internal _, c) | Video (Internal _, c) ->
        txt ~verbatim:false ~in_source:false [ c ]
        @ if in_source then [] else [ Break Paragraph ]
```

Reply:
Sorry, I haven't seen you needed "file" for this...

Reply:
I have lifted the corresponding code to separate auxiliary function to avoid the duplication.